### PR TITLE
config_tool: set MAX_VM_NUM value range from 1 to 8 and validate vm_num

### PR DESCRIPTION
**Pull request by Weiyi-Feng**
1. set MAX_VM_NUM value range from 1 to 8.
2. validate vm_num must <= MAX_VM_NUM at export scenarion xml

Tracked-On: #6685
Signed-off-by: Weiyi Feng <weiyix.feng@intel.com>

### CODE REVIEW

**Review by jsun26intel — APPROVED** [review]

**Comment by wenlingz** [comment]
No new violations to the coding guideline detected.
No New Name Conflict
No New Function Declaration/Definition Mismatch

**Comment by gvancuts** [comment]
Is 8 a hard limit?

I'm thinking about customers who may want to try with more than 8 VMs, does this force them to patch the config tool to try it out?

**Comment by jsun26intel** [comment]
@gvancuts How many VMs that ACRN could run depends on memory layout under 4GB. More VMs means huger hypervisor binary. Some special board only has ~1G space under 4GB.  If bootloader could not put hypervisor binary and kernel modules under 4GB, the boot will fail.
8 VMs is more generic for most boards could run and so far no requirement on this. If users want to run more than 8 VMs, better patch the config tool by themselves and might be risk to boot.

**Comment by wenlingz** [comment]
OK to verify

**Comment by wenlingz** [comment]
OK to verify

**Review by NanlinXie — APPROVED** [review]

File: misc/config_tools/schema/types.xsd [hunk 1]
```diff
@@ -50,6 +50,16 @@
...
 
+<xs:simpleType name="MaxVMNumType">
+  <xs:annotation>
+    <xs:documentation>Integer from 1 to 8.</xs:documentation>
+  </xs:annotation>
+  <xs:restriction base="xs:integer">
+    <xs:minInclusive value="1" />
+    <xs:maxInclusive value="8" />
+  </xs:restriction>
+</xs:simpleType>
```
Comment (junjiemao1) [inline misc/config_tools/schema/types.xsd:61]: Creating a separate integer type solely due to a different range seems an overkill to me. You can embed the type in where it is used directly. See https://www.w3schools.com/xml/el_restriction.asp for examples.

We may need to clean up the types here, but in a future PR probably.

**Pull request merged**